dotnet/src/Connectors/Connectors.OpenAI/AzureSdk/ClientCore.cs

```diff
@@ -323,18 +343,34 @@ await foreach (Completions completions in response.ConfigureAwait(false))
         for (int requestIndex = 1; ; requestIndex++)
         {
             // Make the request.
-            var responseData = (await RunRequestAsync(() => this.Client.GetChatCompletionsAsync(chatOptions, cancellationToken)).ConfigureAwait(false)).Value;
-            this.CaptureUsageDetails(responseData.Usage);
-            if (responseData.Choices.Count == 0)
+            ChatCompletions responseData;
+            IEnumerable<OpenAIChatMessageContent> responseContent;
+            using (var activity = ModelDiagnostics.StartCompletionActivity(this.Endpoint, this.DeploymentOrModelName, ModelProvider, chat, executionSettings))
             {
-                throw new KernelException("Chat completions not found");
+                try
+                {
+                    responseData = (await RunRequestAsync(() => this.Client.GetChatCompletionsAsync(chatOptions, cancellationToken)).ConfigureAwait(false)).Value;
+                    this.CaptureUsageDetails(responseData.Usage);
+                    if (responseData.Choices.Count == 0)
+                    {
+                        throw new KernelException("Chat completions not found");
+                    }
+                }
+                catch (Exception ex)
+                {
+                    activity?.SetError(ex);
+                    throw;
+                }
+
+                responseContent = responseData.Choices.Select(chatChoice => this.GetChatMessage(chatChoice, responseData));
+                activity?.SetCompletionResponse(responseContent, responseData.Usage.PromptTokens, responseData.Usage.CompletionTokens);
```

Comment:
do we handle the case when

- Streaming starts
- N chunks are received
- Connection drops and stream interrupts
?

We need to make sure to 
- report an activity with reasonable error.type
- flush received parts of the stream and report events for the partial context
- probably set finish reason to an error

Reply:
Streaming will be my focus next. This one is for non-streaming APIs to keep the PRs relatively small.